src/kernel/arch/x86/keyboard.zig

```diff
@@ -0,0 +1,140 @@
+const builtin = @import("builtin");
+const build_options = @import("build_options");
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+const irq = @import("irq.zig");
+const arch = if (builtin.is_test) @import(build_options.arch_mock_path ++ "arch_mock.zig") else @import("arch.zig");
+const log = if (builtin.is_test) @import(build_options.arch_mock_path ++ "log_mock.zig") else @import("../../log.zig");
+const panic = @import("../../panic.zig").panic;
+const Keyboard = @import("../../keyboard.zig").Keyboard;
+const KeyPosition = @import("../../keyboard.zig").KeyPosition;
+
+var keyboard: *Keyboard = undefined;
+
+var pressed_keys: usize = 0;
+var special_sequence = false;
+var expected_releases: usize = 0;
+var on_print_screen = false;
+
+///
+/// Register a keyboard action. Should only be called in response to a keyboard IRQ
+///
+/// Arguments:
+///     IN ctx: *arch.CpuState - The state of the CPU when the keyboard action occured
+///
+/// Return: usize
+///     The stack pointer value to use when returning from the interrupt
+///
+fn onKey(ctx: *arch.CpuState) usize {
```

Comment:
Could add unit tests for the special keys by mocking out the `inb` and have it return the sequences for the special keys

Reply:
I thought you'd mention mocking xD

Reply:
XD

Reply:
Mocking this would be very complex so I went with a simpler approach and added a normal test instead.